### PR TITLE
lua, latex - interpret subcells as subfloats when subcap count matches subcell count

### LINKED ISSUE #10328: `tbl-subcap` with LaTeX tables
Minimal repro from `keep-md: true`:

`````qmd
---
format: pdf
keep-tex: true
---

::: {#tbl-example .cell layout-ncol="2" tbl-cap='Example' tbl-subcap='["Cars","Pressure"]'}

```{.r .cell-code}
library(tinytable)
tt(head(cars)) |> theme_tt("tabular") |> 
  save_tt("latex") |> knitr::asis_output()
```

::: {.cell-output-display}
```{=latex}
\begin{tabular}{ll}
\hline
speed & dist \\ \hline
4 &  2 \\
4 & 10 \\
7 &  4 \\
7 & 22 \\
8 & 16 \\
9 & 10 \\
\hline
\end{tabular}
```
:::

```{.r .cell-code}
tt(head(pressure)) |> theme_tt("tabular")  |> 
  save_tt("latex") |> knitr::asis_output()
```

::: {.cell-output-display}
```{=latex}
\begin{tabular}{ll}
\hline
temperature & pressure \\ \hline
0 & 0.0002 \\
20 & 0.0012 \\
40 & 0.0060 \\
60 & 0.0300 \\
80 & 0.0900 \\
100 & 0.2700 \\
\hline
\end{tabular}
```
:::
:::
`````

### Discussed in https://github.com/quarto-dev/quarto-cli/discussions/10324

<div type='discussions-op-text'>

<sup>Originally posted by **vincentarelbundock** July 18, 2024</sup>
### Description

Hi,

Thanks for all your work on Quarto and on this forum!

When a package produces markdown tables, the `tbl-subcap` functionality works as expected. However, when the package produces a LaTeX table, there is no subcaption.

Is it possible for Quarto to support subcaptions in LaTeX tables as well?

I'd love to get some clarification on this.

1. Are subcaptions only supported for markdown tables?
2. Are there plans to extend support for LaTeX tables as well?

Thanks!

````
---
format: pdf
---

```{r}
#| label: tbl-example
#| tbl-cap: "Example"
#| tbl-subcap: 
#|   - "Cars"
#|   - "Pressure"
#| layout-ncol: 2

library(tinytable)
tt(head(cars)) |> save_tt("markdown") |> knitr::asis_output()
tt(head(pressure)) |> save_tt("markdown") |> knitr::asis_output()
```
````

This doesn't work:

```r 
library(tinytable)
tt(head(cars)) |> save_tt("latex") |> knitr::asis_output()
tt(head(pressure)) |> save_tt("latex") |> knitr::asis_output()
```</div>

### PR DESCRIPTION
**Pull request by cscheid**
Closes #10328.

### CODE REVIEW

File: tests/docs/smoke-all/2024/07/18/10324.qmd [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        - ['\\subcaption{\\label{tbl-example-2}Pressure}']
+        - []
+---
+
+::: {#tbl-example .cell layout-ncol="2" tbl-cap='Example' tbl-subcap='["Cars","Pressure"]'}
+
+```{.r .cell-code}
+library(tinytable)
+tt(head(cars)) |> theme_tt("tabular") |> 
+  save_tt("latex") |> knitr::asis_output()
```
Comment (vincentarelbundock) [inline tests/docs/smoke-all/2024/07/18/10324.qmd:17]: This line is superfluous. It was given in a reprex to match another example, but `save_tt()` and `knitr::asis_output()` should be called automatically by `tinytable`.

File: tests/docs/smoke-all/2024/07/18/10324.qmd [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+8 & 16 \\
+9 & 10 \\
+\hline
+\end{tabular}
+```
+:::
+
+```{.r .cell-code}
+tt(head(pressure)) |> theme_tt("tabular")  |> 
+  save_tt("latex") |> knitr::asis_output()
```
Comment (vincentarelbundock) [inline tests/docs/smoke-all/2024/07/18/10324.qmd:38]: Also superfluous.

File: tests/docs/smoke-all/2024/07/18/10324.qmd [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      ensureFileRegexMatches:
+        - ['\\subcaption{\\label{tbl-example-2}Pressure}']
+        - []
+---
+
+::: {#tbl-example .cell layout-ncol="2" tbl-cap='Example' tbl-subcap='["Cars","Pressure"]'}
+
+```{.r .cell-code}
+library(tinytable)
+tt(head(cars)) |> theme_tt("tabular") |>
```
Comment (vincentarelbundock) [inline tests/docs/smoke-all/2024/07/18/10324.qmd:16]: Just a reminder that `theme_tt("tabular")` only works in the development version of `tinytable`. Hopefully this doesn't break your tests.

File: tests/docs/smoke-all/2024/07/18/10324.qmd (hunk 1 above)
Comment (cscheid) [reply tests/docs/smoke-all/2024/07/18/10324.qmd:17]: Got it, and I'll make sure to use that in the future. But note that in this case this is not executable code anyway, it's the _output_ from knitr. We do this in our test suite so we don't have to worry about installing as many dependencies.

File: tests/docs/smoke-all/2024/07/18/10324.qmd (hunk 2 above)
Comment (cscheid) [reply tests/docs/smoke-all/2024/07/18/10324.qmd:38]: See above.

File: tests/docs/smoke-all/2024/07/18/10324.qmd (hunk 3 above)
Comment (cscheid) [reply tests/docs/smoke-all/2024/07/18/10324.qmd:16]: See above :)

**Pull request merged**